java/feathub-connectors/flink-connectors/flink-connector-redis/src/main/java/com/alibaba/feathub/flink/connectors/redis/lookup/RedisLookupFunction.java

```diff
@@ -0,0 +1,191 @@
+
+            String key =
+                    ConversionUtils.toString(
+                            rowData,
+                            i + originalKeyFieldIndices.length,
+                            fieldTypes[redisKeyFieldIndices[i]]);
+            int valueFieldIndex = redisValueFieldIndices[i];
+            DataType valueType = fieldTypes[valueFieldIndex];
+            if (valueType instanceof CollectionDataType) {
+                result.setField(
+                        valueFieldIndex,
+                        ConversionUtils.fromList(
+                                client.lrange(key, 0, -1), (CollectionDataType) valueType));
+            } else if (valueType instanceof KeyValueDataType) {
+                result.setField(
+                        valueFieldIndex,
+                        ConversionUtils.fromMap(client.hgetAll(key), (KeyValueDataType) valueType));
+            } else {
+                result.setField(
+                        valueFieldIndex, ConversionUtils.fromString(client.get(key), valueType));
+            }
+        }
+
+        return Collections.singletonList(result);
+    }
+
+    @Override
+    public void close() throws Exception {
+        super.close();
+        this.client.close();
+    }
+
+    private static int[] getOriginalKeyFieldIndices(ResolvedSchema schema, ReadableConfig config) {
+        String[] keyFields = config.get(KEY_FIELDS).split(",");
+        List<Integer> indices = new ArrayList<>();
+        List<String> fieldNames = schema.getColumnNames();
+        for (String keyField : keyFields) {
+            int keyFieldIndex = fieldNames.indexOf(keyField);
+            Preconditions.checkArgument(
+                    keyFieldIndex >= 0, "Input table does not contain key field %s.", keyField);
+            indices.add(keyFieldIndex);
+        }
+        return indices.stream().mapToInt(Integer::intValue).toArray();
+    }
+
+    private static int[] getRedisKeyFieldIndices(ResolvedSchema schema, int[] valueFieldIndices) {
+        List<Integer> indices = new ArrayList<>();
+        List<String> fieldNames = schema.getColumnNames();
+        for (int i : valueFieldIndices) {
+            int keyFieldIndex = fieldNames.indexOf("__KEY__" + fieldNames.get(i));
+            Preconditions.checkArgument(
+                    keyFieldIndex >= 0,
+                    "Input table does not contain key field %s.",
+                    "__KEY__" + fieldNames.get(i));
+            indices.add(keyFieldIndex);
+        }
+        return indices.stream().mapToInt(Integer::intValue).toArray();
+    }
+
+    private static int[] getRedisValueFieldIndices(ResolvedSchema schema, ReadableConfig config) {
+        List<String> keyFields = Arrays.asList(config.get(KEY_FIELDS).split(","));
+        List<Integer> indices = new ArrayList<>();
+        List<String> fieldNames = schema.getColumnNames();
+        for (int i = 0; i < schema.getColumnCount(); i++) {
```

Comment:
It might be more readable to replace `schema.getColumnCount()` with `fieldNames.size()`.